src/arraytypes/primitive.jl

```diff
@@ -34,7 +34,7 @@ Base.size(p::Primitive) = (p.ℓ,)
 
 function Base.copy(p::Primitive)
     if nullcount(p) == 0
```

Comment:
```suggestion
function Base.copy(p::Primitive{T, A}) where {T, A}
    if nullcount(p) == 0 && T === eltype(A)
```
I think this should allow us to keep the optimization for non-converting `Primitive` eltypes.